### ISSUE TITLE
Office.onReady()/Office.initialize is not running in Excel online add-in using shared js runtime and Office.addin.setStartupBehavior(Office.StartupBehavior.load)

### ISSUE THREAD

**Issue by ivark16** [OfficeDev/office-js#1371]
I'm creating an add-in in Excel where I want to detect changes in the workbook when the taskpane is not open and run custom code when the document is reopened / reloaded. 

I have a button in my add-in where the user can set start up behaviour to load which triggers this code:

`Office.addin.setStartupBehavior(Office.StartupBehavior.load);`

I have tried using both `Office.onReady()` and` Office.initialize` but neither one is invoked when the document is reloaded after the `setStartupBehavior` has been set to load. 

I have followed the Microsoft documentation on these matters but still have a problem in Excel online. 

Documentation that I have been following:
[Configure your Excel add-in to use a shared JavaScript runtime](https://docs.microsoft.com/en-us/office/dev/add-ins/excel/configure-your-add-in-to-use-a-shared-runtime)
[Run code in your Excel add-in when the document opens](https://docs.microsoft.com/en-us/office/dev/add-ins/excel/run-code-on-document-open)

Please note that this function works in the Excel desktop app but not in Excel online. Also, this add-in is still in dev mode and has not been published live. 

## Expected Behavior

That Office.onReady() / Office.initialize should run afther the `Office.addin.setStartupBehavior(Office.StartupBehavior.load);` parameters have been set when reloading / reopening the document. It should print out into the console "Hello world!".

## Current Behavior

After setting the `Office.addin.setStartupBehavior(Office.StartupBehavior.load);` parameters the `onReady()` / `initialize` is never called on document open unless the taskpane of the add-in is opened which is not the expected behaviour. 

## Steps to Reproduce, or Live Example

* Additional details:

1. yo Office
2. Add to manifest.xml (As described in the "Configure your Excel add-in to use a shared JavaScript runtime" documentation)
`<Runtimes>
        <Runtime resid="Taskpane.Url" lifetime="long" />
</Runtimes>`

3. Add to package.json under dev dependencies
"@types/office-js-preview": "^1.0.85",

4. npm i

5. add to index.tsx Office.initialize()
`Office.addin.setStartupBehavior(Office.StartupBehavior.load);
console.log("Hello world!");`

6. npm run dev-server

7. Sideload AddIn into browser: Insert -> Add-ins -> Manage My Add-ins-> Upload Manifest.xml

8. Open sidePane to trigger Office.initialize which sets startupBeheaviour to load

9. Reload document

## Context

We are trying to create an Excel add-in which can be used both online and on the desktop client. It detects changes in the Excel worksheet even when the add-in is not open or loaded. For starters this add-in will only be using the taskpane and run code on reload / refresh.  

## Your Environment

* Platform [PC desktop, Mac, iOS, Office on the web]: Office on the web
* Host [Excel, Word, PowerPoint, etc.]: Excel
* Office version number: 
* Operating System: Windows
* Browser (if using Office on the web): Microsoft Edge / Google Chrome

**Comment by AlexJerabek** [comment]
Thanks for reporting this @ivark16. 

@jeremy-msft or @shaofengzhu, could you please help with this shared runtime issue?

**Comment by jeremy-msft** [comment]
Thanks for your report @ivark16, we'll take a look and get back to you soon.

**Comment by ElizabethSamuel-MSFT** [comment]
@jeremy-msft Any update here? Thx.

**Comment by jeremy-msft** [comment]
We are able to reproduce the issue and currently investigating a fix.

**Comment by sshukurov** [comment]
We are experiencing the same issue. Calling `Office.addin.setStartupBehavior(Office.StartupBehavior.load)` outputs the following error to console: **RichApi.Error: This operation is not implemented.**
![image](https://user-images.githubusercontent.com/36232686/94925954-7870ba00-04d9-11eb-9f10-e15d1d97ab42.png)

**Comment by ivark16** [comment]
Hi, are there any updates on this issue?

**Comment by jeremy-msft** [comment]
Hi @ivark16, thanks for your patience.  This should now be fixed; please let us know if you continue to see the problem.

**Comment by ivark16** [comment]
It seems to be working now, thank you guys so much! 😃

**Comment by jeremy-msft** [comment]
You're welcome, we're glad we were able to resolve this!

**Comment by ivark16** [comment]
Hi @jeremy-msft,

I have run into this issues again. After deploying my add-in code to Azure App Service I noticed this problem has resurfaced. I have tried with the same code running both locally (running the code on localhost) and live from Azure App Service. It all works when run locally but the bug is still present when hosting the add-in on azure.

**Comment by jeremy-msft** [comment]
Hi @ivark16, we were not able to repro your issue using a sideloaded manifest pointed to a non-localhost server.  Can you please provide additional details about your environment and any console errors you are seeing?   Is the server URL the only difference between your two manifests?  Does it work fine on the Desktop client?  Also, please let us know the build number shown in Online under File -> About.

**Comment by ivark16** [comment]
Hi @jeremy-msft,

Thanks so much for your answer. I decided to take a closer look at this problem. The manifest file that is producing the bug is distributed by centralized deployment through the admin center [(via this documentation)](https://docs.microsoft.com/en-us/office/dev/add-ins/publish/centralized-deployment).  I decided to take my existing manifest file that references my live Azure code and insert it manually as an Add-in in Excel online instead (the same file that is distributed by centralized deployment).
![image](https://user-images.githubusercontent.com/24226040/100366452-bd582f00-2ff8-11eb-847d-6dd7ddbeb653.png)

By doing that I was not able to reproduce the bug, everything worked as it should. So that got me thinking if the bug is linked to distributing the add-in via centralized deployment? That is the only difference I can think of. 

Regarding your other questions: 
- There are no console errors that I'm noticing. 
- Yes the server URL is the only difference between the manifest files.
- Yes this works fine on the Desktop client, as it always has. 
-  Excel online build number: 16.0.13524.31875
